packages/json-api/src/Registry.ts

```diff
@@ -96,12 +96,32 @@ export default class RegistryImpl implements Registry {
         }, {})
       : undefined;
 
-    if (Array.isArray(document.data)) {
-      return (document.data as JSONAPIResource[]).map(elem =>
-        parseResource(elem, includesMap)
+    let included: { [key: string]: Array<JSONValue> } = {};
+    if (includesMap) {
+      included = Object.keys(includesMap).reduce(
+        (acc: { [key: string]: Array<JSONValue> }, key) => {
```

Comment:
This type is used is several places, so it probably deserves its own name (which could be local to this file).

Reply:
👍